addons/project/tests/test_project_flow.py

```diff
@@ -550,3 +550,20 @@ def test_copy_project_with_embedded_actions(self):
         self.assertEqual(duplicated_task_embedded_filter.embedded_parent_res_id, new_project_pigs.id)
         self.assertEqual(duplicated_task_embedded_filter.action_id, task_embedded_filter.action_id)
         self.assertEqual(duplicated_task_embedded_filter.model_id, task_embedded_filter.model_id)
+
+    def test_mail_on_task_project_change(self):
+        """
+            This test will check that a mail is sent to the subscribers having the
+            "Task created" follow subtype when the project_id of said task changes
+        """
+        subtype_id = self.env['ir.model.data']._xmlid_to_res_id('project.mt_project_task_new')
+        partner = self.env['res.partner'].search([('user_ids', '=', self.user_projectuser.id)])
+        self.env['mail.followers'].create({
+            'res_model': 'project.project',
+            'res_id': self.project_goats.id,
+            'partner_id': partner.id,
+            'subtype_ids': [subtype_id],  # task created subtype
+        })  # set user as follower of project
+        with self.mock_mail_gateway():
+            self.task_1.project_id = self.project_goats.id  # change project
```

Comment:
equivalent to:

```suggestion
            self.task_1.project_id = self.project_goats
```

(no need to add the comment since it is really obvious we change the project. 😇)